### ISSUE TITLE
Start shipping Mono in .app

### ISSUE THREAD

**Issue by leafi** [leafi/fix-spacechem-os-x#3]
http://www.mono-project.com/news/2016/03/31/mono-relicensed-mit/

https://github.com/mono/mono/blob/master/LICENSE

With the more liberal relicense of Mono as a result of MS' Xamarin purchase, AFAICT we can start shipping Mono Mono in the .app bundle without issue. (Someone tell me if I'm wrong!)

The user no longer having to download Mono themselves should solve an entire class of problems, as well as making it easier on (obviously) the user.

**Comment by leafi** [comment]
Things that don't work, but keeping note in case we do want to just ship one massive binary one day:

(Issue is SetCurrentDirectory in .net ends up being called, but with a mysteriously empty path (apparently)).

`AS="as -arch i386" CC="cc -arch i386 -framework CoreFoundation -lobjc -liconv" PKG_CONFIG_PATH=/Library/Frameworks/Mono.framework/Libraries/pkgconfig MONO_PATH=. mkbundle --static --deps SpaceChem.exe -o testo`

(linking against corefoundation et al. not needed <4.4)

**Comment by leafi** [comment]
Things that seem to be working:

MonoKickstart based on... a random commit very shortly after 4.4.0, courtesy of https://github.com/MadMatt04/MonoKickstart/ . Using this branch because this guy has hacked out 64-bit support, presumably for something like the reason we need to. (Otherwise libnative.dylib won't load for us - it's 32-bit only...)

Stole I18N\* DLLs from my native OS X 4.4.0 installation to solve ICSharp Zip stuff causing Mono to complain it doesn't understand codepage IBM437.

Currently dealing with the 4.4 Web problems. I believe Mono introduced support for having the System.Web.\* stack backed by native stuff going through P/Invoke or the old DIY stack, and this is the root cause of the problems. Rather than try and figure out how to pass the correct configuration - because, in all likelihood, with kick.c there IS NO WAY TO PASS THE CORRECT CONFIGURATION - I'm currently installing 4.3.2.a-little-bit and planning to steal me some DLLs.

**Comment by leafi** [comment]
Alright. Spent ages messing around with it... I haven't fixed the web stuff, but the web stuff I saw in the console was actually just a crash upload failing, so I've been working through the crash logs instead.

Managed to get replacement SDL crap in _roughly_ the right place. Hey, it's in the .app - that's good enough for me right now.

Lots of redundant files. Going to refresh SpaceChem.app from Steam and transplant back in.

Current problems:
- Web shit broken (might well end up just reverse engineering what parts of the web request whatever SpaceChem uses, and writing a shim)
- Why doesn't System.Data.Sqlite work?!?
- Will probably need to shove libgdiplus.dylib in. But the various MonoKickstart repositories have this, so whatever.
- I have completely lost track of which files I got from where, and will probably have to reconstruct my work once I've got a POC
